docs/design_proposals/ko-builder.md

```diff
@@ -0,0 +1,240 @@
+# ko builder
+
+* Author(s): Halvard Skogsrud (@halvards)
+* Design Shepherd: \<skaffold-core-team-member\>
+* Date: 2021-03-29
+* Status: Draft
+
+## Objectives
+
+Fast, standardized, reproducible, configuration-less, Docker-less, and
+secure-by-default container image builds for Go apps.
+
+## Background
+
+[ko](https://github.com/google/ko) is a container image builder for Go. It's
+[fast](https://cloud.google.com/blog/topics/developers-practitioners/ship-your-go-applications-faster-cloud-run-ko),
+doesn't use a `Dockerfile` or rely on the Docker daemon, and uses
+[distroless](https://github.com/GoogleContainerTools/distroless) base images by
+default. ko is to Go apps what
+[Jib](https://github.com/GoogleContainerTools/jib) is to JVM-based apps
+(approximately).
+
+The [Knative](https://knative.dev/) and [Tekton](https://tekton.dev/) open
+source projects use ko.
+
+## Proposal
+
+This proposal adds a new `ko` builder to Skaffold, based on the `ko publish`
+command. The integration does _not_ include other ko functionality related to
+[rendering](https://github.com/google/ko#ko-resolve) manifests,
+[deploying](https://github.com/google/ko#ko-apply) to Kubernetes clusters, and
+[file watching](https://github.com/google/ko/blob/f7df8106196518df5c6c35432843421e33990329/pkg/commands/resolver.go#L240).
+
+Compared to ...
+
+- [the Cloud Native buildpacks builder](https://skaffold.dev/docs/pipeline-stages/builders/buildpacks/),
+  the ko builder is
+  [fast](https://cloud.google.com/blog/topics/developers-practitioners/ship-your-go-applications-faster-cloud-run-ko),
```

Comment:
great numbers! 
Users can still achieve this by custom script right? I think the main selling point is what you have listed in the next section which is no additional tool set up in CI or on dev machine. 

It would be help us rationalize this effort a lot with numbers.
I see `ko` uses a user agent when publishing an image. It would be easy for us to find how many images are pushed on GCR with ko.  Reach out to me internally or take a look at go/container-tools-metrics/kaniko to kaniko usage.